src/main/java/com/emc/mongoose/metrics/context/DistributedMetricsContextImpl.java

```diff
@@ -1,28 +1,31 @@
-package com.emc.mongoose.metrics;
-
-import com.codahale.metrics.Snapshot;
-import com.codahale.metrics.UniformSnapshot;
+package com.emc.mongoose.metrics.context;
 
 import com.emc.mongoose.item.op.OpType;
-
+import com.emc.mongoose.metrics.DistributedMetricsListener;
+import com.emc.mongoose.metrics.DistributedMetricsSnapshotImpl;
+import com.emc.mongoose.metrics.MetricsSnapshot;
+import com.emc.mongoose.metrics.util.Meter;
+import com.emc.mongoose.metrics.util.RateMetricSnapshot;
+import com.emc.mongoose.metrics.util.RateMetricSnapshotImpl;
+import com.emc.mongoose.metrics.util.TimingMetricSnapshot;
+import com.emc.mongoose.metrics.util.TimingMetricSnapshotImpl;
 import com.github.akurilov.commons.system.SizeInBytes;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 import java.util.function.IntSupplier;
 import java.util.function.Supplier;
 
 public class DistributedMetricsContextImpl<S extends DistributedMetricsSnapshotImpl>
-extends MetricsContextBase<S>
-implements DistributedMetricsContext<S> {
+	extends MetricsContextBase<S>
+	implements DistributedMetricsContext<S> {
 
 	private final IntSupplier nodeCountSupplier;
 	private final Supplier<List<MetricsSnapshot>> snapshotsSupplier;
 	private final boolean avgPersistFlag;
 	private final boolean sumPersistFlag;
 	private final boolean perfDbResultsFileFlag;
-
-	private volatile long prevElapsedTime = 0;
 	private volatile DistributedMetricsListener metricsListener = null;
```

Comment:
failed to find any usage of the metrics listener
why?

Reply:
I don't know what listener was used for. It was in MeterMBean...